### PR TITLE
Improvements for MultiplayerScreen

### PR DESCRIPTION
**Pull request by GGGuenni**
+ Fixed ResponsePopup: It's now working like intended

Improvements:
+ "Add Current Game" Button now kills itself
+ "Next Turn" updates the multiplayer list
+ After deleting the currently running game the MultiplayerScreen now shows the "Add Current Game" Button
+ MultiplayerScreen loads a lot faster
+ Clicking "Add Current Game" can no longer freeze the game
+ Renaming a multiplayer game to an already used name no longer overrides the save file

### CODE REVIEW

File: core/src/com/unciv/ui/MultiplayerScreen.kt [hunk 1]
```diff
@@ -223,71 +199,126 @@ class MultiplayerScreen() : PickerScreen() {
...
-                //skipping one save is not fatal
-                ResponsePopup("Could not refresh!".tr(), this)
-                continue
             }
+            leftSideTable.clear()
+            leftSideTable.add(leftSubTable)
+        }
+    }
+
+    /* Is this worth the translation effort? Seems definitely more polished if used
```
Comment (GGGuenni) [inline core/src/com/unciv/ui/MultiplayerScreen.kt]: What do you think?
I don't really like to add to many new translation strings 😬

**Comment by yairm210** [comment]
To many changes at once, I'm not following.

A. What is the problem were trying to solve, and what is the proposed solution?
B. Once problem at a time - it looks like you're dealing 'along the way' with other issues, which should be separated into different PRs. 

I'm only human, I can't cope with this :)

**Comment by GGGuenni** [comment]
Yeah sorry, it's definitely a bit much and could have been split.
I just tried to clean everything up to my likings as fast as possible so I could start working on the multiple games support for the turn checker.
So I first tried to tackle a problem I had when opening the multiplayer screen. It took 2-3 seconds to load the screen on older devices, which was very annoying for me.
My first approach was to add the MultiplayerFileManager to get a clearer view of the code. I just thought having save logic inside a UI class is not a clever thing to do.
I rewrote the MultiplayerScreen to use the MultiplayerFileManager when I realized a few not so polished looking things.
Changing them on the way made it hard to follow so the UX improvements should have been a different PR.
I can close this one and open two new ones one after the other if you want.

**Comment by yairm210** [comment]
My other problem is that your multiplayer manager seems to be static object inside the game saver, and there are places where this make the distinction between them very unclear.
In my opinion, if it's meant to be a id:filename cache, then it should only have caching related functions, and the file saving and loading should be done in the existing class

**Comment by yairm210** [comment]
And yes, splitting it up would be helpful :)
